src/Quizzer.API.App/Controllers/AnswerController.cs

```diff
@@ -0,0 +1,79 @@
+using Microsoft.AspNetCore.Mvc;
+using Quizzer.API.BL.Facades;
+using Quizzer.API.BL.Models;
+
+namespace Quizzer.API.Controllers {
+    [ApiController]
+    [Route("api/answers")]
+    public class AnswerController : ControllerBase {
+        private readonly ILogger<AnswerController> _logger;
+        private readonly AnswerFacade _answerFacade;
+
+        public AnswerController(ILogger<AnswerController> logger,
+            AnswerFacade answerFacade) {
+            _logger = logger;
+            _answerFacade = answerFacade;
+        }
+
+        [HttpGet("", Name = "GetAll")]
+        public IEnumerable<AnswerDetailModel> GetAll() {
+            return _answerFacade.GetAll();
+        }
+
+        [HttpGet("{id:guid}", Name = "GetById")]
+        [ProducesResponseType(StatusCodes.Status404NotFound, Type = typeof(Guid))]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(AnswerDetailModel))]
+        public IActionResult GetById(Guid id) {
+            AnswerDetailModel answer = _answerFacade.GetDetailById(id);
+            
+            if (answer.Id != id)
+                return NotFound(id);
+      
+            return Ok(answer);
+        }
+
+        [HttpGet("byQuestion/{questionId:guid}", Name = "GetForQuestion")]
+        public IEnumerable<AnswerDetailModel> GetForQuestion(Guid questionId) {
+            return _answerFacade.GetAnswersForQuestion(questionId);
+        }
+
+        [HttpPut("", Name = "UpdateAnswer")]
+        [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(AnswerDetailModel))]
+        [ProducesResponseType(StatusCodes.Status400BadRequest, Type = typeof(string))]
+        public IActionResult UpdateAnswer([FromBody] AnswerDetailModel answerModel) {
+            Guid updatedId;
+            try {
+                updatedId = _answerFacade.Save(answerModel);
+            } catch (Exception ex) {
+                return BadRequest(ex.Message);
+            }
+            return Ok(updatedId);
+        }
```

Comment:
@BRTEK007 @KoubaFromPucks  I would like to discuss here about whether it is good idea to separate update and insert or make it inside one call, because I created facades in a way where Save combines update and insert

Reply:
I separated it from the API standpoint. So if you want to add something new you send POST request, and PUT for updating. If we have it together we must choose one.

Reply:
> I separated it from the API standpoint. So if you want to add something new you send POST request, and PUT for updating. If we have it together we must choose one.

@BRTEK007 is right it would be better to do it separatly

Reply:
@BRTEK007 yes it makes sense from the http request standpoint.